### PR TITLE
Add hook to allow dockerfile preprocessing

### PR DESCRIPTION
**Pull request by srlowe**
Change-type: minor
Signed-off-by: Scott Lowe <scott@balena.io>

### CODE REVIEW

**Comment by srlowe** [comment]
@balena-ci retest

File: lib/build-task.ts [hunk 1]
```diff
@@ -124,4 +124,10 @@ export interface BuildTask {
 	 * The container contract for this service
 	 */
 	contract?: Dictionary<unknown>;
+
+	/**
+	 * Promise to ensure that build task is resolved before
+	 * progressing to next steps in the build process (workaround).
+	 */
+	resolvedPromise?: Promise<any>;
```
Comment (CameronDiver) [inline lib/build-task.ts:132]:
```suggestion
	resolvedPromise?: Promise<void>;
```

**Comment by srlowe** [comment]
@balena-ci retest

**Comment by srlowe** [comment]
@balena-ci retest

**Comment by srlowe** [comment]
@balena-ci retest

**Pull request merged**